crates/m3/src/gadgets/hash/keccak/mod.rs

```diff
@@ -79,6 +80,9 @@ pub struct Keccakf {
 	///
 	/// This is used for the state-in to state-out linking rule.
 	next_state_in: StateMatrix<PackedLane8>,
+
+	pub input: StateMatrix<Col<B64>>,
+	pub output: StateMatrix<Col<B64>>,
```

Comment:
Nit: `next_state_in` and `link_sel` are very related and by declaring those here you split them apart. I'd suggest moving those decls top or below.

On another note, since those are public interface, the user might very appreciate if you left a doc about those.